packages/middleware-sdk-s3/src/validate-bucket-name.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { validateBucketName } from "./validate-bucket-name";
+import { SerializeHandler } from "@aws-sdk/types";
+
+describe("validateBucketName", () => {
+  const mockNextHandler = jest.fn();
+  const composedHandler: SerializeHandler<any, any> = validateBucketName(
+    mockNextHandler
+  );
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("throws error if Bucket parameter contains '/'", async () => {
+    const bucket = "bucket/part/of/key";
+    let error;
+    try {
+      await composedHandler({
```

Comment:
The following can be stored in a parameter say `args`
```javascript
const args = {
  input: {
    Bucket: bucket
  }
}
```

This way you can add following check at the end of the file:
```javascript
expect(mockNextHandler.mock.calls[0][0]).toBe(args);
```